packages/grafana-ui/src/components/IconButton/IconButton.tsx

```diff
@@ -15,76 +15,92 @@ export type IconButtonVariant = 'primary' | 'secondary' | 'destructive';
+}
+
+interface BasePropsWithTooltip extends BaseProps {
+  /** Tooltip content to display on hover and as the aria-label */
+  tooltip: PopoverContent;
+  /** Position of the tooltip */
+  tooltipPlacement?: TooltipPlacement;
+}
+
+interface BasePropsWithAriaLabel extends BaseProps {
+  /** @deprecated use aria-label instead*/
   ariaLabel?: string;
+  /** Text available only for screen readers. No tooltip will be set in this case. */
+  ['aria-label']: string;
 }
 
-export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
-  (
-    {
-      name,
-      size = 'md',
-      iconType,
-      tooltip,
-      tooltipPlacement,
-      ariaLabel,
-      className,
-      variant = 'secondary',
-      ...restProps
-    },
-    ref
-  ) => {
-    const theme = useTheme2();
-    let limitedIconSize: LimitedIconSize;
-
-    // very large icons (xl to xxxl) are unified to size xl
-    if (size === 'xxl' || size === 'xxxl') {
-      deprecationWarning('IconButton', 'size="xxl" and size="xxxl"', 'size="xl"');
-      limitedIconSize = 'xl';
-    } else {
-      limitedIconSize = size;
-    }
-
-    const styles = getStyles(theme, limitedIconSize, variant);
-    const tooltipString = typeof tooltip === 'string' ? tooltip : '';
-
-    // When using tooltip, ref is forwarded to Tooltip component instead for https://github.com/grafana/grafana/issues/65632
-    const button = (
+export type Props = BasePropsWithTooltip | BasePropsWithAriaLabel;
+
+export const IconButton = React.forwardRef<HTMLButtonElement, Props>((props, ref) => {
+  const { size = 'md', variant = 'secondary' } = props;
+
+  const theme = useTheme2();
+  let limitedIconSize: LimitedIconSize;
+
+  // very large icons (xl to xxxl) are unified to size xl
+  if (size === 'xxl' || size === 'xxxl') {
+    deprecationWarning('IconButton', 'size="xxl" and size="xxxl"', 'size="xl"');
+    limitedIconSize = 'xl';
+  } else {
+    limitedIconSize = size;
+  }
+
+  const styles = getStyles(theme, limitedIconSize, variant);
+
+  let ariaLabel: string | undefined;
+  let buttonRef: typeof ref | undefined;
+
+  if ('tooltip' in props) {
+    const { tooltip } = props;
+    ariaLabel = typeof tooltip === 'string' ? tooltip : undefined;
+  } else if ('ariaLabel' in props || 'aria-label' in props) {
+    const { ariaLabel: deprecatedAriaLabel, ['aria-label']: ariaLabelProp } = props;
+    ariaLabel = ariaLabelProp || deprecatedAriaLabel;
+    buttonRef = ref;
+  }
+
+  // When using tooltip, ref is forwarded to Tooltip component instead for https://github.com/grafana/grafana/issues/65632
+  if ('tooltip' in props) {
+    const { name, iconType, className, tooltip, tooltipPlacement, ...restProps } = props;
+    return (
+      <Tooltip ref={ref} content={tooltip} placement={tooltipPlacement}>
+        <button
+          {...restProps}
+          ref={buttonRef}
+          aria-label={ariaLabel}
+          className={cx(styles.button, className)}
+          type="button"
+        >
```

Comment:
This button seems to be the same as the one in line 92 (unless I'm missing something). What is the reason why we changed it to duplicate the code instead of just declaring the button and using it in the if/else like we had before?